extensions/ql-vscode/src/variant-analysis/view-autofixes.ts

```diff
@@ -0,0 +1,886 @@
+  const queryFilePathNoExt = join(
+    parsedQueryFilePath.dir,
+    parsedQueryFilePath.name,
+  );
+
+  // Get the path to the query help, which may be either a `.qhelp` or a `.md` file.
+  // Note: we assume that the name of the query file is the same as the name of the query help file.
+  const queryHelpFilePathQhelp = `${queryFilePathNoExt}.qhelp`;
+  const queryHelpFilePathMarkdown = `${queryFilePathNoExt}.md`;
+
+  // Set `queryHelpFilePath` to the existing extension type.
+  let queryHelpFilePath: string;
+  if (await pathExists(queryHelpFilePathQhelp)) {
+    queryHelpFilePath = queryHelpFilePathQhelp;
+  } else if (await pathExists(queryHelpFilePathMarkdown)) {
+    queryHelpFilePath = queryHelpFilePathMarkdown;
+  } else {
+    throw new Error(
+      `Could not find query help file at either ${queryHelpFilePathQhelp} or ${queryHelpFilePathMarkdown}. Check that the query help file exists and is named correctly.`,
+    );
+  }
+
+  // Get the query metadata.
+  const metadata = await tryGetQueryMetadata(cliServer, queryFilePath);
+  if (!metadata) {
+    throw new Error(`Could not get query metadata for ${queryFilePath}.`);
+  }
+  // Get the query ID (used for the overridden query help's filename).
+  const queryId = metadata.id;
+  if (!queryId) {
+    throw new Error(`Query metadata for ${queryFilePath} is missing an ID.`);
+  }
+  // Replace `/` with `-` for use with the overridden query help's filename.
+  // Use `replaceAll` since some query IDs have multiple slashes.
+  const queryIdWithDash = queryId.replaceAll("/", "-");
+
+  // Get the path to the output directory for overriding the query help.
+  // Note: the path to this directory may change in the future.
+  const queryHelpOverrideDirectory = join(
+    localAutofixPath,
+    "prompt-templates",
+    "qhelps",
+    `${queryIdWithDash}.md`,
+  );
+
+  await cliServer.generateQueryHelp(
+    queryHelpFilePath,
+    queryHelpOverrideDirectory,
+  );
+}
+
+/**
+ * Gets the full names (owner/repo) of the selected
+ * repositories from the given variant analysis while
+ * limiting the number of repositories to `MAX_NUM_REPOS`.
+ */
+function getSelectedRepositoryNames(
+  variantAnalysis: VariantAnalysis,
+  filterSort: RepositoriesFilterSortStateWithIds,
+): string[] {
+  // Get the repositories that were selected by the user.
+  const filteredRepositories = filterAndSortRepositoriesWithResults(
+    variantAnalysis.scannedRepos,
+    filterSort,
+  );
+
+  // Get the full names (owner/repo = nwo) of the selected repos.
+  let fullNames = filteredRepositories
+    ?.filter((a) => a.resultCount && a.resultCount > 0)
+    .map((a) => a.repository.fullName);
+  if (!fullNames || fullNames.length === 0) {
+    throw new Error("No repositories with results found.");
+  }
+
+  // Limit to MAX_NUM_REPOS by slicing the array.
+  if (fullNames.length > MAX_NUM_REPOS) {
+    fullNames = fullNames.slice(0, MAX_NUM_REPOS);
+    void extLogger.log(
```

Comment:
Should this perhaps be a `window.showWarningMessage` call so that the user is notified more directly?